packages/kit/src/runtime/app/navigation.js

```diff
@@ -101,6 +101,20 @@ export const preloadCode = /* @__PURE__ */ client_method('preload_code');
  */
 export const beforeNavigate = /* @__PURE__ */ client_method('before_navigate');
 
+/**
+ * A lifecycle function that runs the supplied `callback` immediately before we navigate to a new URL.
+ *
+ * If you return a Promise, SvelteKit will wait for it to resolve before completing the navigation. This allows you to — for example — use `document.startViewTransition`. Avoid promises that are slow to resolve, since navigation will appear stalled to the user.
```

Comment:
```suggestion
 * If you return a `Promise`, SvelteKit will wait for it to resolve before completing the navigation. This allows you to — for example — use `document.startViewTransition`. Avoid promises that are slow to resolve, since navigation will appear stalled to the user.
```